sampuru-server/pom.xml

```diff
@@ -39,8 +39,8 @@
         <artifactId>maven-compiler-plugin</artifactId>
         <version>3.8.1</version>
         <configuration>
-          <source>14</source>
-          <target>14</target>
+          <source>17</source>
+          <target>17</target>
```

Comment:
Java 17 is set here using `source` and `target` parameters, but it's set to 14 in the parent pom via the `release` parameter. The `release` parameter effectively does the same thing as `source`+`target`. I'm not sure what result this config will actually have, but 14 definitely shouldn't be specified anywhere.

Since there is a parent pom, I think best practice would be to set up the compiler plugin there and remove it from the child, but either way works since there's only one module.